Assets/Scripts/Statistics/Trackers/PatrollingTracker.cs

```diff
@@ -33,8 +38,9 @@ public sealed class PatrollingTracker : Tracker<PatrollingVisualizer, IPatrollin
         //TODO: TotalCycles is not set any where in the code
         public int TotalCycles { get; }
 
-        public readonly List<PatrollingSnapShot> SnapShots = new();
-        public readonly Dictionary<Vector2Int, List<WaypointSnapShot>> WaypointSnapShots;
+        private readonly BlockingCollection<(PatrollingSnapshot, WaypointSnapshot[])> _snapshots = new();
+
```

Comment:
Readability suggestion: Move it to line 44, such that all the stuff about snapshots are located together 😄